MAVProxy/modules/mavproxy_system_time.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+'''
+System_Time Module
+Peter barker, May 2018
+
+Send mavlink SYSTEM_TIME messages via mavlink
+
+'''
+
+import os
+import os.path
+import sys
+from pymavlink import mavutil
+import errno
+import time
+
+from MAVProxy.modules.lib import mp_module
+from MAVProxy.modules.lib import mp_util
+from MAVProxy.modules.lib import mp_settings
+
+
+class system_time(mp_module.MPModule):
+    def __init__(self, mpstate):
+        """Initialise module"""
+        super(system_time, self).__init__(mpstate, "system_time", "")
+        self.last_sent = 0
+
+        self.system_time_settings = mp_settings.MPSettings(
+            [ ('verbose', bool, False),
+              ('interval', int, 10),
+            ])
+        self.add_command('system_time', self.cmd_system_time, "system_time module", ['status','set (LOGSETTING)'])
+
+    def usage(self):
+        '''show help on command line options'''
+        return "Usage: system_time <status|set>"
+
+    def cmd_system_time(self, args):
+        '''control behaviour of the module'''
+        if len(args) == 0:
+            print(self.usage())
+        elif args[0] == "status":
+            print(self.status())
+        elif args[0] == "set":
+            self.system_time_settings.command(args[1:])
+        else:
+            print(self.usage())
+
+    def status(self):
+        '''returns information about module'''
+        return "Perfectly happy.  All is good with the world"
+
+    def uptime(self):
+        ret = 0
+        try:
+            with open("/proc/uptime") as f:
```

Comment:
This won't work on Windows, right? Isn't there a proper Python API?

Reply:
Correct - won't work in Windows. There should be something in the pywin32 package to help with this.